wurst/lib/InventoryEvent.wurst

```diff
@@ -0,0 +1,98 @@
+/*
+*  InventoryEvent v1.0.0.0
+*     by Bannar
+*
+*  For intuitive inventory event handling.
+*/
+package InventoryEvent
+import RegisterEvents
+
+unit eventUnit = null
+item eventItem = null
+int eventSlotFrom = -1
+int eventSlotTo = -1
+trigger eventMovedTrigger = CreateTrigger()
+trigger eventUsedTrigger = CreateTrigger()
+
+public constant EVENT_INVENTORY_MOVED = 0
+public constant EVENT_INVENTORY_USED = 1
+
+public function getInventoryManipulatingUnit() returns unit
+    return eventUnit
+
+public function getInventoryManipulatedItem() returns item
+    return eventItem
+
+public function getInventorySlotFrom() returns int
+    return eventSlotFrom
+
+public function getInventorySlotTo() returns int
+    return eventSlotTo
+
+public function getInventorySwappedItem() returns item
+    return UnitItemInSlot(eventUnit, eventSlotTo)
+
+public function unit.getItemSlot(item itm) returns int
+    if this.hasItem(itm)
+        let size = this.inventorySize()
+        for i = 0 to size-1
+            if this.itemInSlot(i) == itm
+                return i
+    return -1
+
+function fireEvent(trigger evt, unit u, item itm, integer slotFrom, integer slotTo)
+    unit prevUnit = eventUnit
+    item prevItem = eventItem
+    int prevSlotFrom = eventSlotFrom
+    int prevSlotTo = eventSlotTo
+
+    eventUnit = u
+    eventItem = itm
+    eventSlotFrom = slotFrom
+    eventSlotTo = slotTo
+
+    evt.evaluate()
+
+    eventUnit = prevUnit
+    eventItem = prevItem
+    eventSlotFrom = prevSlotFrom
+    eventSlotTo = prevSlotTo
+
+function onItemOrder()
+    let order = GetIssuedOrderId()
+    let u = GetTriggerUnit()
+    item itm
+    int slotFrom
+    int slotTo
+
+    if order >= 852002 and order <= 852007
+        itm = GetOrderTargetItem()
+        slotFrom = u.getItemSlot(itm)
+        slotTo = order - 852002
+        fireEvent(eventMovedTrigger, u, itm, slotFrom, slotTo)
+    else
+        slotFrom = order - 852008
+        itm = u.itemInSlot(slotFrom)
+        fireEvent(eventUsedTrigger, u, itm, slotFrom, -1)
+
+function onAnyOrder()
+    let order = GetIssuedOrderId()
+    if order >= 852002 and order <= 852013
```

Comment:
Please either comment magic numbers or place them into descriptive variables (possibly both for particularly complicated ones).

Reply:
Issuing orders by string is slower than by ints.
And there is no complication, because in most cases you just use few of them and you are not creator of library. User should deal with exposed interface, not play with internal structures.

When playing with orders, you can have a order-id list open, such as: https://github.com/nestharus/JASS/blob/master/jass/Systems/OrderIds/script.j

Reply:
Makes sense, but to @jonathanmclaus' point can we do something like

```
if order >= ORDER_moveslot1 and order <= ORDER_useslot6
```

Once the constant-inliner runs it will become the same thing.

Reply:
Nice! Didin't know wurst has the vjass equivalent there.